invenio_jobs/tasks.py

```diff
@@ -52,6 +76,10 @@ def execute_run(self, run_id, kwargs=None):
         )
         raise e
     except (TaskExecutionPartialError, TaskExecutionError) as e:
+        current_app.logger.error(
+            f"Run {run.id} encountered an error: {e.message} with Sentry Event ID: {getattr(g, 'sentry_event_id', None)}"
+        )
```

Comment:
minor: `sentry_event_id` is fetched below as well, but the bigger issue is that it might not be there at all for most InvenioRDM instances, so we should always be adding it in the error message.

Also I would add the sentry ID in the logged message as an `extra`, so that it's an optional data field in the job logs that the UI can conditionally display